lib/utils/display.js

```diff
@@ -82,26 +82,31 @@ const ERROR_KEY = 'error'
 // is a json error that should be merged into the finished output
 const JSON_ERROR_KEY = 'jsonError'
 
+const isPlainObject = (v) => v && typeof v === 'object' && !Array.isArray(v)
+
 const getArrayOrObject = (items) => {
-  // Arrays cant be merged, if the first item is an array return that
-  if (Array.isArray(items[0])) {
-    return items[0]
-  }
-  // We use objects with 0,1,2,etc keys to merge array
-  if (items.length && items.every((o, i) => Object.hasOwn(o, i))) {
-    return Object.assign([], ...items)
+  if (items.length) {
+    const foundNonObject = items.find(o => !isPlainObject(o))
+    // Non-objects and arrays cant be merged, so just return the first item
+    if (foundNonObject) {
+      return foundNonObject
+    }
+    // We use objects with 0,1,2,etc keys to merge array
+    if (items.every((o, i) => Object.hasOwn(o, i))) {
+      return Object.assign([], ...items)
+    }
   }
-  // Otherwise its an object with all items merged together
-  return Object.assign({}, ...items)
+  // Otherwise its an object with all object items merged together
+  return Object.assign({}, ...items.filter(o => isPlainObject(o)))
```

Comment:
```suggestion
  return { ...items.filter(o => isPlainObject(o)) }
```
not a huge deal, but this should be slightly faster

Reply:
items is an array of objects, so i dont think it can be spread within another object in the same way as arguments to a function

```js
> Object.assign({}, ...[{a:1},{b:2}])
{ a: 1, b: 2 }
> { ...[{a:1},{b:2}] }
{ '0': { a: 1 }, '1': { b: 2 } }
```

Reply:
ahhh you're totally right